coalib/core/CircularDependencyError.py

```diff
@@ -12,8 +13,24 @@ def __init__(self, node=None):
             The node that was encountered twice and closes the dependency
             circle.
         """
-        message = 'Circular dependency detected.'
+        prefix = 'Circular dependency detected'
         if node is not None:
-            message += ' {0!r} -> ... -> {0!r}'.format(node)
+            message = '{prefix}: {node!r} -> ... -> {node!r}'.format(
+                prefix=prefix, node=node)
+        else:
+            message = prefix + '.'
+        return cls(message)
 
-        Exception.__init__(self, message)
+    @classmethod
+    def for_bears(cls, bears):
+        """
+        Creates the CircularDependencyError with a helpful message about the
+        dependency.
+
+        :param bears:
+            sequence of bears, for which a circular dependency was detected.
+        """
+        bear_names = [bear.name for bear in bears]
+
+        return cls('Circular dependency detected: {names}'.format(
+            names=' -> '.join(bear_names)))
```

Comment:
What about having only a constructor like before, but use this code instead? And the core which currently wants to print `node1 -> ... -> node2` passes `bear1object, '...', bear2object`.
-->
```python
def __init__(nodes):
        prefix = 'Circular dependency detected'
        if nodes is not None:
            message = '{}: {}'.format(prefix, ' -> '.join(str(node) for node in nodes)
        else:
            message = prefix + '.'

        Exception.__init__(self, message)
```
The collection stuff just uses the constructor normally by feeding in the bear names `(bear.name for bear in bears)` and the core would feed in this: `[repr(node), '...', repr(node)]`